validates.js

```diff
@@ -1,67 +1,93 @@
 import React from 'react';
+import { string, func, element, oneOf } from 'prop-types';
+
+const undef = void 0;
+function noop() {}
 
 export default class Validates extends React.Component {
+  get propsHandler() {
+    const { onValidChange = noop } = this.props;
+    return onValidChange;
+  }
+
+  get ctxHandler() {
+    const { onValidChange = noop } = this.context;
+    return onValidChange;
+  }
+
   /**
-   * If isValid !== wasValid, calls the onValidChange handlers in props and context with the specified arguments.
+   * If `isValid !== wasValid` or `oldName !== this.props.name`, calls the onValidChange handlers in props and context with the
+   * specified arguments.
    *
-   * @param {Mixed} isValid Validity. `true`/`false` if the component is valid/invalid; `null` if validation is disabled
-   * @param {Mixed} wasValid The previous validity. May be `undefined` if this is the first update.
+   * @param {Mixed} isValid Validity. `true`/`false` if the component is valid/invalid; `null` if validation is disabled;
+   * `undefined` if there is no validation at all.
+   * @param {Mixed} wasValid The previous validity.
+   * @param {Object} [oldName] The old name that the component was using. If it is different than the current name, the props and
+   * context handlers will be called first with `undefined` to indicate th old name no longer has validation.
    */
-  onValidChange (isValid, wasValid) {
-    if (isValid === wasValid) {
-      return;
-    }
-
-    const { onValidChange: propsHandler, name } = this.props;
-    const { onValidChange: ctxHandler } = this.context;
+  onValidChange(isValid, wasValid, oldName) {
```

Comment:
Nitpick: rename `oldName` to `prevName` for consistency with other rename nitpicks. Nitpicks on nitpicks?